codeguru_profiler_agent/sdk_reporter/sdk_reporter.py

```diff
@@ -99,52 +104,53 @@ def report(self, profile):
-    def create_pg_when_one_click_integration_is_active(self):
+    @with_timer("createProfilingGroup", measurement="wall-clock-time")
+    def create_profiling_group(self):
         """
         Create a PG for the Lambda function onboarded with 1-click integration
         """
-        function_name = str(self.metadata.fleet_info.function_arn).split(':')[6]
-
-        profiling_group_name_for_1_click = "aws-lambda-{}".format(function_name)
-        profiling_group_name_to_be_created = os.getenv("AWS_CODEGURU_PROFILER_GROUP_NAME",
-                                                       profiling_group_name_for_1_click)
         try:
             self.codeguru_client_builder.codeguru_client.create_profiling_group(
-                profilingGroupName=profiling_group_name_to_be_created,
+                profilingGroupName=self.profiling_group_name,
                 computePlatform='AWSLambda'
             )
-            self.profiling_group_name = profiling_group_name_to_be_created
-            self.is_lambda_one_click_pg_created_during_execution = True
-            logger.info("Created Lambda Profiling Group with name {}", profiling_group_name_to_be_created)
-
+            self.is_profiling_group_created_during_execution = True
+            logger.info("Created Lambda Profiling Group with name " + str(self.profiling_group_name))
         except Exception as e:
             self._log_request_failed(operation="create_profiling_group", exception=e)
 
-    def is_lambda_one_click_integration_active(self):
+    def should_autocreate_profiling_group(self):
+        return self.is_runtime_python() and self.is_handler_name_env_variable_set()
+
+    def is_handler_name_env_variable_set(self):
         """
         Check if the ComputeType is AWSLambda and if the environment
         variables for Lambda Layer Profiling are set
         """
-        fleet_info_as_map = self.metadata.fleet_info.serialize_to_map()
-        if 'computeType' in fleet_info_as_map:
-            if fleet_info_as_map['computeType'] == 'aws_lambda':
-                handler_name_ = os.environ.get(HANDLER_ENV_NAME_FOR_CODEGURU)
-                if not handler_name_:
-                    logger.error("Env Variables for CodeGuru Profiler Lambda Layer are not set. Cannot create PG.")
-                    return False
-                return True
-        return False
+        handler_name_ = os.environ.get(HANDLER_ENV_NAME_FOR_CODEGURU_KEY)
+        if not handler_name_:
+            logger.info("Env Variables for CodeGuru Profiler Lambda Layer are not set. Cannot create Profiling Group.")
+            return False
+        return True
+
+    def is_runtime_python(self):
+        """
+        Check if the runtime for the AWS Lambda function is Python
+        """
+        # https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html
+        python_runtime_identifier_prefix = "python"
+        execution_environment_env_variable = os.environ.get(AWS_EXECUTION_ENV_KEY)
```

Comment:
What would be the lambda's runtime to be non-python, but this code to run?

Reply:
As per above comment! Why do we need to know if the execution environment is python or not?

Reply:
yeah my bad not needed. removed it